src/Common/filesystemHelpers.cpp

```diff
@@ -234,35 +234,30 @@ bool createFile(const std::string & path)
     DB::throwFromErrnoWithPath("Cannot create file: " + path, path, DB::ErrorCodes::CANNOT_CREATE_FILE);
 }
 
+bool exists(const std::string & path)
+{
+    return faccessat(AT_FDCWD, path.c_str(), F_OK, AT_EACCESS) == 0;
+}
+
 bool canRead(const std::string & path)
 {
-    struct stat st;
-    if (stat(path.c_str(), &st) == 0)
-    {
-        if (st.st_uid == geteuid())
-            return (st.st_mode & S_IRUSR) != 0;
-        else if (st.st_gid == getegid())
-            return (st.st_mode & S_IRGRP) != 0;
-        else
-            return (st.st_mode & S_IROTH) != 0 || geteuid() == 0;
-    }
+    if(exists(path))
+        return faccessat(AT_FDCWD, path.c_str(), R_OK, AT_EACCESS) == 0;
```

Comment:
I refactored all the `stat` syscalls to use `faccessat` instead since it should be faster because it makes fewer kernel lookups operations.

Reply:
I would prefer to make this in separate pull request, I expect that this could be an issue for our intergation tests.